cli/cmd/staking.go

```diff
@@ -56,19 +61,63 @@ func newCreateValCmd() *cobra.Command {
 	return cmd
 }
 
+// eoaConfig defines the required data to sign and submit evm transactions.
+type eoaConfig struct {
+	network        netconf.ID
+	executionRPC   string
+	consensusRPC   string
+	privateKeyFile string
+}
+
+func (v eoaConfig) privateKey() (*ecdsa.PrivateKey, error) {
+	if v.privateKeyFile == "" {
+		return nil, errors.New("required flag --private-key-file not set")
+	}
+
+	opPrivKey, err := crypto.LoadECDSA(v.privateKeyFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "load private key")
+	}
+
+	return opPrivKey, nil
+}
+
+func (v eoaConfig) validate() error {
+	if _, err := v.privateKey(); err != nil {
+		return errors.Wrap(err, "verify --private-key-file flag")
+	}
+
+	if err := v.network.Verify(); err != nil {
+		return errors.Wrap(err, "verify --network flag")
+	}
+
+	if v.executionRPC != "" {
+		if _, err := url.Parse(v.executionRPC); err != nil {
+			return errors.Wrap(err, "verify --execution-rpc flag")
+		}
+	}
+
+	if v.consensusRPC != "" {
+		if _, err := url.Parse(v.consensusRPC); err != nil {
+			return errors.Wrap(err, "verify --consensus-rpc flag")
+		}
+	}
+
+	return nil
+}
+
 type createValConfig struct {
-	Network            netconf.ID
-	PrivateKeyFile     string
-	ConsensusPubKeyHex string
-	SelfDelegation     uint64
+	eoaConfig
+	consensusPubKeyHex string
+	selfDelegation     uint64
 }
```

Comment:
I know most people say that ALL fields of private types must also be private. But I like to design and communiate the API of a type, even if it is a private type. I therefore use exported fields to indicate that these fields can be read/written by any function. And that they are not private to the type. Private fields should never be accessed by another outside the type. They should be accessed via getters/setters. 

Also, simple structs with exported fields indicate simple structs. I'd suggest leaving the fields as public to make that explicit.